src/lib/block.h

```diff
@@ -43,10 +43,9 @@ struct block_device;
  *                                @see block_device::cntrl_path.
  * @param[in]    ibpi             an IBPI pattern (state) to visualize.
  *
- * @return 1 if successful, otherwise the function returns 0.
+ * @return STATUS_SUCCESS if successful, otherwise a valid status_t status code.
  */
-typedef int (*send_message_t) (struct block_device *device,
-			       enum led_ibpi_pattern ibpi);
+typedef status_t (*send_message_t) (struct block_device *device, enum led_ibpi_pattern ibpi);
```

Comment:
please remove the typdef and put it directly in struct in following form:

```C
status_t (*send_message_t)(struct block_device *device, enum led_ibpi_pattern ibpi);
```

the typedefing like that make reader experience worse.

you can also update flush function, but it is not mandatory :)

it will break _get_send_fn but we don't need assigned to `send_message_t result = NULL` ther we can pass block device and assign function directly. I know that it is not purpose of your change but it should be smooth and nice improvement!

Reply:
done